docs/authentication.md

```diff
@@ -32,35 +34,50 @@ The Helios client will now need to present a client certificate to authenticate.
+   **Note that this step must be implemented by the cluster administrator.**
 
-3. To prove that you own the corresponding private key, the Helios client asks
-   ssh-agent to cryptographically hash and sign part of the TLS handshake with
-   the server.
+When using the Helios CLI the username in the generated certificate comes from the `--username` argument, which defaults to your login user. When using the API the username has to be set explicitly via the `HeliosClient.Builder.setUser()` method.
 
-The hashing and signing in the final step is a standard part of the TLS handshake
-and how the client proves to the server that it has possession of the private key.
-The only difference in our implementation is that we ask ssh-agent to do the
-signing, since Helios doesn't have access to your private key.
+The actual handshake procedure is carried out by the TLS implementation -
+the only thing the Helios client does it generate and sign the X.509 certificate used, instead
+of using an existing certificate.
 
-**It is important to note that the generated certificate is a self-signed
-certificate**, and all it proves is that you have the private key corresponding
+**It is important to note that the generated certificate is signed only by your
+SSH key**, and verifying its signature only proves that you have the private key corresponding
 to your SSH public key. The web server must still decide whether or not to accept
 your certificate. At Spotify, we've configured nginx to pass the certificate to
 another tiny service which extracts the public key from the certificate and
 verifies that it matches the public SSH key of one of our employees.
+
+### Authenticating with the `helios` CLI
+
+When using the CLI there are two ways to supply a certificate to the server when needed:
+
+ * Set the `HELIOS_CERT_PATH` environment variable to a path containing the pre-issued X.509 certificate and corresponding private key, named `cert.pem` and `key.pem` respectively, or
+ * Let the Helios client generate a temporary X.509 certificate that is signed with your private SSH key via ssh-agent, as described above.
+
+The latter is typically not suitable for non-interactive use cases (e.g. a script run periodically by cron). If the `HELIOS_CERT_PATH` environment variable is set this method takes precedence.
+
+### Authenticating with the Helios API
+
+When you use the Helios API (via the `HeliosClient` class) you have the same options as when using the CLI, as explained above. You can also explicitly set the path where the certificate and key is via the `HeliosClient.Builder.setClientCertificatePath()` method. If a certificate path is set explicitly the `HELIOS_CERT_PATH` is ignored.
+
+Again, for non-interactive use cases relying on ssh-agent is typically not a suitable solution so we recommend that you specify a path either explicitly or via the `HELIOS_CERT_PATH` environment variable.
```

Comment:
Thinking about it a little more, it probably doesn't make sense to give an exampled detail here -- how you should do it will depend on the specific setup. E.g. we use self-signed certs and maintain a list of trusted certs, but other's might have proper PKI setup.